### PR TITLE
WFCORE-640 - Add file permissions checks during execution of add-user.bat/add-user.sh

### PR DESCRIPTION
**Pull request by bgaisford**
Corrected the issue in the unit test case, this is good to go now.

### CODE REVIEW

**Comment by darranl** [comment]
Generally I think this looks good, the only thing I would say is that the StateValues class is more for the propagation of values from state to state as the utility executes - when we are within a single state an alternative should possibly be used even if that does mean some refactoring of the method calls within the state.

File: domain-management/src/test/java/org/jboss/as/domain/management/security/adduser/PropertyFileFinderPermissionsTestCase.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.domain.management.security.adduser;
+
+import org.jboss.as.domain.management.security.adduser.AddUser.FileMode;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.*;
```
Comment (darranl) [inline domain-management/src/test/java/org/jboss/as/domain/management/security/adduser/PropertyFileFinderPermissionsTestCase.java]: Please fully expand the import to cover the classes actually required.

**Comment by bgaisford** [comment]
Would you like me to refactor PropertyFileFinder as per your comments above?  Happy to do so.

**Comment by darranl** [comment]
Yes please, try and keep it so StateValues is only used for values that need to transition to other states.

**Comment by bstansberry** [comment]
this is ok to test

**Comment by bgaisford** [comment]
This is good to go Darran.  It was as simple as moving the permissions checks state from StateValues to internal members of PropertyFileFinder.  Thanks for the excellent feedback.  Please let me know if there is anything else you would like changed based on the latest updates.

**Comment by kabir** [comment]
@bgaisford before this can be merged, please answer the following question:

We require that all contributions be made under the terms of the MIT License, http://www.opensource.org/licenses/mit-license.php. Do you agree to the these terms?

**Comment by bgaisford** [comment]
Yes, I agree to the terms of the MIT License.

**Comment by bgaisford** [comment]
**@kabir** During development I did create a new source file and I copied the license from an existing similar source file.  Upon closer inspection, the license text I used (see below) doesn't match the license you sent in the link above.  Should I update the source file with the MIT license?
"/*
 * JBoss, Home of Professional Open Source.
 * Copyright 2012, Red Hat, Inc., and individual contributors
 * as indicated by the @author tags. See the copyright.txt file in the
 * distribution for a full listing of individual contributors.
 *
 * This is free software; you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as
 * published by the Free Software Foundation; either version 2.1 of
 * the License, or (at your option) any later version.
 *
 * This software is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this software; if not, write to the Free
 * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
 * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 */"

**Comment by kabir** [comment]
@bgaisford The license header is the correct one, although set the year to 2015 rather than 2012.

The MIT license agreement is more about your contributions, and being able to relicense them:
"Until now, contributor agreements applicable to WildFly have generally limited outbound licensing of contributions under LGPL (or  GPL) only. We would now like to have the flexibility to consider  other outbound open source licensing options for components of WildFly that may be incompatible with or less restrictive than LGPL  licensing."

**Comment by bgaisford** [comment]
@kabir Understood.  I have verified licenses and rev'd copyright year in touched and new files.

**Comment by kabir-ncl** [comment]
@bgaisford sorry to be a pain. But you should only update the year in your new files, not in the existing ones which you have modified.

**Comment by kabir** [comment]
@bgaisford sorry to be a pain. But you should only update the year in your new files, not in the existing ones which you have modified.

**Comment by bgaisford** [comment]
@kabir Done, no worries.

**Pull request merged**